grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -271,4 +271,68 @@ class CycleCountService {
         }
         return cycleCounts.collect { CycleCountDto.createFromCycleCountItems(it.cycleCountItems) }
     }
+
+    CycleCountDto submitCount(CycleCountSubmitCountCommand command) {
+        command.cycleCount.cycleCountItems.each { CycleCountItem cycleCountItem ->
+            updateCycleCountItemForSubmit(cycleCountItem, command.refreshQuantityOnHand, command.failOnOutdatedQuantity)
+            determineCycleCountItemStatus(cycleCountItem, command.requireRecountOnDiscrepancy)
+        }
+        if (command.cycleCount.status == CycleCountStatus.COUNTED) {
+            createCycleCountTransaction(command.cycleCount)
+        }
+        return CycleCountDto.createFromCycleCountItems(command.cycleCount.cycleCountItems)
+    }
+
+    void updateCycleCountItemForSubmit(CycleCountItem cycleCountItem, boolean refreshQuantityOnHand, boolean failOnOutdatedQuantity) {
+        Integer currentQuantityOnHand =
+            productAvailabilityService.getQuantityOnHandInBinLocation(cycleCountItem.inventoryItem, cycleCountItem.location)
+        if (refreshQuantityOnHand) {
+            cycleCountItem.quantityOnHand = currentQuantityOnHand
+        }
+        if (failOnOutdatedQuantity && cycleCountItem.quantityOnHand != currentQuantityOnHand) {
```

Comment:
this will always be false if refreshQuantityOnHand is true. You'll want to check if currentQuantityOnHand != to  cycleCountItem.quantityOnHand BEFORE you've updated it (ie if the old value has changed).

the idea is that if the QoH is changing for the product we want the user to review it again to make sure everything looks as they expected. (This will only technically be true for recounts for us because all our regular counts are blind counts and so don't need to be reviewed, but I chose to be able to support it for regular counts as well.)